shell/app/modules/msp/pages/micro-service-overview/index.tsx

```diff
@@ -12,56 +13,270 @@
-                  {item.name}
-                </span>
-              </RadioButton>
-            );
-          })}
-        </RadioGroup>
+    <div className="flex flex-1 flex-col min-h-0 bg-white shadow pb-2">
+      <div className="px-4 py-2 bg-header flex justify-between">
+        <Input
+          prefix={<ErdaIcon color="gray" type="search" />}
+          bordered={false}
+          allowClear
+          placeholder={i18n.t('msp:search by project name')}
+          className="bg-hover-gray-bg w-72"
+          onChange={(e) => {
+            handleSearch(e.target.value);
+          }}
+        />
+        <ErdaIcon size="20" type="refresh" onClick={onReload} />
+      </div>
+      <div className="px-2 mt-2 flex-1 overflow-y-auto">
+        <Spin spinning={dataLoading}>
+          {serviceList?.length ? (
+            serviceList.map(({ id, language, views, lastHeartbeat, name }) => {
+              return (
+                <Row
+                  key={id}
+                  className="project-item card-shadow mb-2 mx-2 px-4 flex py-6 rounded-sm"
+                  onClick={() => {}}
+                >
+                  <Col span={10} className="flex items-center">
+                    <div className="w-14 h-14 mr-2">
+                      <ErdaIcon type={ERDA_ICON[language]} size="56" />
+                    </div>
+                    <div>
+                      <p className="mb-0.5 font-medium text-xl leading-8">{name}</p>
+                      <Tag color="#59516C" className="mb-0.5 text-xs leading-5 border-0">
+                        {i18n.t('msp:last active time')}: {lastHeartbeat}
+                      </Tag>
+                    </div>
+                  </Col>
+                  <Col span={14} className="flex items-center">
+                    <Row gutter={8} className="flex-1">
+                      {map(views, ({ data, type, view }) => {
+                        const timeStamp: number[] = [];
+                        const value: number[] = [];
+                        view.map((item) => {
+                          timeStamp.push(item.timestamp);
+                          value.push(item.value);
+                        });
+                        const currentOption = {
+                          ...option,
+                          xAxis: { data: timeStamp, show: false },
+                          series: [
+                            {
+                              ...defaultSeriesConfig,
+                              data: value,
+                              type: 'line',
+                            },
+                          ],
+                        };
+
+                        const errorRateOption = {
+                          ...option,
+                          xAxis: { data: timeStamp, show: false },
+                          series: [
+                            {
+                              ...defaultSeriesConfig(value.find((val) => val !== 0) ? '#d84b65' : ''),
```

Comment:
defaultSeriesConfig  is function  or variable？？

Reply:
modified